sql/src/main/java/io/crate/operation/projectors/ShardDMLExecutor.java

```diff
@@ -171,13 +157,15 @@ private String getLocalNodeId(ClusterService clusterService) {
         return nodeId;
     }
 
-    private void processShardResponse(ShardResponse shardResponse) {
-        if (shardResponse.failure() != null) {
-            executionFuture.completeExceptionally(shardResponse.failure());
-            return;
+    private BitSet processShardResponse(ShardResponse shardResponse) {
+        Exception failure = shardResponse.failure();
+        if (failure != null) {
+            Throwables.throwIfUnchecked(failure);
+            throw new RuntimeException(failure);
```

Comment:
Is this better somehow? Why not fail the execution immediately? (is it worth creating a new exception here?)